app/models/authentication.rb

```diff
@@ -24,4 +25,11 @@ def set_source
                        resource.try(:source_id)
                      end
   end
+
+  def only_one_superkey
+    superkey_authtype = source.source_type.superkey_authtype
+    return unless superkey_authtype && authtype == superkey_authtype
+
+    source.authentications.count { |auth| auth.authtype == superkey_authtype } > 1
```

Comment:
What if `source.authentications.count { |auth| auth.authtype == superkey_authtype }` was equal to 3 or 5?  Wouldn't this still return true?

Maybe that would be fine - it's just that the method name stipulates `only_one_superkey`.

Reply:
Updated it to be a bit more readable, using `.any?` rather than `> 1`.